jsmn_stream_token.c

```diff
@@ -0,0 +1,278 @@
+#include "jsmn_stream_token.h"
+#include <stdbool.h>
+
+static jsmn_streamtok_t *jsmn_stream_allocate_token(jsmn_stream_token_parser_t *jsmn_stream_parser);
+static jsmn_streamtok_t *jsmn_stream_get_super_token(jsmn_stream_token_parser_t *jsmn_stream_parser);
+static void jsmn_stream_parse_tokens_start_array(void *user_arg);
+static void jsmn_stream_parse_tokens_end_array(void *user_arg);
+static void jsmn_stream_parse_tokens_start_object(void *user_arg);
+static void jsmn_stream_parse_tokens_end_object(void *user_arg);
+static void jsmn_stream_parse_tokens_object_key(const char *key, size_t key_length, void *user_arg);
+static void jsmn_stream_parse_tokens_string(const char *value, size_t length, void *user_arg);
+static void jsmn_stream_parse_tokens_primitive(const char *value, size_t length, void *user_arg);
+
+static jsmn_stream_callbacks_t jsmn_stream_token_callbacks = {
+	.start_array_callback = jsmn_stream_parse_tokens_start_array,
+	.end_array_callback = jsmn_stream_parse_tokens_end_array,
+	.start_object_callback = jsmn_stream_parse_tokens_start_object,
+	.end_object_callback = jsmn_stream_parse_tokens_end_object,
+	.object_key_callback = jsmn_stream_parse_tokens_object_key,
+	.string_callback = jsmn_stream_parse_tokens_string,
+	.primitive_callback = jsmn_stream_parse_tokens_primitive
+};
+
+/**
+ * @brief Initialize the jsmn_stream_token_parser_t object.
```

Comment:
It would be helpful to include information about the expected usage of `jsmn_stream_parser`, `tokens`, and `num_tokens` in the function's comments.